x/concentrated-liquidity/internal/math/tick_test.go

```diff
@@ -49,20 +110,52 @@ func (suite *ConcentratedMathTestSuite) TestPriceToTick() {
 	testCases := []struct {
 		name         string
 		price        sdk.Dec
+		kAtPriceOne  sdk.Int
 		tickExpected string
 	}{
 		{
-			"happy path",
-			sdk.NewDec(5000),
-			"85176",
+			"50000 to tick with -4 k at price one",
+			sdk.NewDec(50000),
+			sdk.NewInt(-4),
+			"400000",
+		},
+		{
+			"5.01 to tick with -2 k at price one",
+			sdk.MustNewDecFromStr("5.010000000000000000"),
+			sdk.NewInt(-2),
+			"401",
+		},
+		{
+			"50000.01 to tick with -6 k at price one",
+			sdk.MustNewDecFromStr("50000.010000000000000000"),
+			sdk.NewInt(-6),
+			"40000001",
+		},
+		{
+			"0.00000889 to tick with -8 k at price one",
+			sdk.MustNewDecFromStr("0.000008890000000000"),
+			sdk.NewInt(-8),
+			"-99999111",
+		},
+		{
+			"0.9998 to tick with -4 k at price one",
+			sdk.MustNewDecFromStr("0.999800000000000000"),
+			sdk.NewInt(-4),
+			"-2",
+		},
+		{
```

Comment:
Let's add a test at max spot price please and the minimum possible kAtPriceOne

Reply:
[c1178b9](https://github.com/osmosis-labs/osmosis/pull/3874/commits/c1178b905516e8bf76e58e8e94bde65e9d5b1b34)